src/wallet/rpcdump.cpp

```diff
@@ -185,6 +185,8 @@ UniValue importprivkey(const JSONRPCRequest& request)
                 pwallet->ImportScripts({GetScriptForDestination(WitnessV0KeyHash(vchAddress))}, 0 /* timestamp */);
             }
         }
+        // Scan mempool for transactions
+        pwallet->chain().requestMempoolTransactions(*pwallet);
```

Comment:
Why is this done even when rescan is turned off? I think this should only be run when rescan is on.

If the user turns off rescan, they know that
* either no transaction has happened in the past, so rescan is not needed
* or they do a full rescan later

This also means that this should be added to the rescan RPC.

Reply:
> This also means that this should be added to the rescan RPC.

Might be the best approach, it would cover any import that rescans and also manual rescan. So in practice `ScanForWalletTransactions` where `stop_block.IsNull` would scan mempool too?

Reply:
> So in practice `ScanForWalletTransactions` where `stop_block.IsNull` would scan mempool too?

@MarcoFalke is this your suggestion?